torchgeo/trainers/detection.py

```diff
@@ -44,13 +44,24 @@
 class ObjectDetectionTask(pl.LightningModule):
     """LightningModule for object detection of images.
 
+    Currently, supports a Faster R-CNN model from
+    `torchvision
+    <https://pytorch.org/vision/stable/models/faster_rcnn.html>`_ with
+    one of the following *backbone* arguments:
+
+    .. code-block:: python
+
+        ['resnet18', 'resnet34', 'resnet50', 'resnet101', 'resnet152',
+        'resnext50_32x4d','resnext101_32x8d', 'wide_resnet50_2',
+        'wide_resnet101_2']
```

Comment:
Did we want to use timm here too or is that not easy to do? It would allow us to ignore all the weights craziness above.

Reply:
Agreed, but not sure if it works to have any timm backbone with a torchvision detection architecture, maybe @ashnair1 knows or has an opinion? Otherwise, I have time to test it tomorrow.

Reply:
segmentation-models-pytorch doesn't support object detection, does it?

Reply:
Plugging in timm backbones into torchvision detectors would be difficult.  https://github.com/pytorch/vision/issues/6172 links to an implementation of Faster RCNN with a timm backbone but it looks like a lot of work was needed for it to function. 

> segmentation-models-pytorch doesn't support object detection, does it?

No

Reply:
Let's stick with what we have for now and work on improvements if we find a better option in the future.